src/toil/common.py

```diff
@@ -463,6 +477,63 @@ def __hash__(self) -> int:
         return self.__dict__.__hash__()  # type: ignore
 
 
+def check_and_create_toil_home_dir() -> None:
+    """
+    Ensure that TOIL_HOME_DIR exists.
+
+    Raises an error if it does not exist and cannot be created. Safe to run
+    simultaneously in multiple processes.
+    """
+
+    dir_path = try_path(TOIL_HOME_DIR)
+    if dir_path is None:
+        raise RuntimeError(f"Cannot create or access Toil configuration directory {TOIL_HOME_DIR}")
+
+def generate_config(filepath: str) -> None:
+    """
+    Write a Toil config file to the given path.
+
+    Safe to run simultaneously in multiple processes. No process will see an
+    empty or partially-written file at the given path.
+    """
+    # this is placed in common.py rather than toilConfig.py to prevent circular imports
+
+    # configargparse's write_config does not write options with a None value, so do this instead
+    omit = ("help", "config", "defaultAccelerators", "nodeTypes", "nodeStorageOverrides", "setEnv", "minNodes", "maxNodes", "logCritical", "logDebug", "logError", "logInfo", "logOff", "logWarning", "linkImports", "noLinkImports", "moveExports", "noMoveExports", "disableCaching")
```

Comment:
I don't really understand what "this" is. Why do these have to be excluded from the default config? Since we're just dumping YAML ourselves, why do we care what their `write_config` does? Are we trying to mimic it for some reason?

Instead of just listing facts about some options here (which will change when the options are added/removed/adjusted), can't we look at the stored default value instead to detect the options we want to skip?